superset/utils/core.py

```diff
@@ -940,15 +940,20 @@ def get_since_until(form_data):
                 since = today
                 until = today + relativedelta(**{grain: int(num)})
     else:
-        since = form_data.get('since', '')
+        since = since or ''
         if since:
             since = add_ago_to_since(since)
         since = parse_human_datetime(since)
-        until = parse_human_datetime(form_data.get('until', 'now'))
+        until = parse_human_datetime(until or 'now')
 
     return since, until
 
 
+def assert_from_to_dttm(from_dttm, to_dttm):
```

Comment:
NIT (optional): to me there's no need to refactor out a 2 line methods into the `utils/` package if it's used in a single place in the code base. At that point it's just unnecessary indirection.

Reply:
What do you think of computing `from_dttm` and `to_dttm` and asserting their validity as part of `get_since_util` (i.e. moving the logic in the blob below into `get_since_util`). We'll provide time_shift as an additional argument to get_since_until. 
```
        self.from_dttm = None if since_dttm is None else (since_dttm - time_shift_dttm)
        self.to_dttm = None if until_dttm is None else (until_dttm - time_shift_dttm)
        utils.assert_from_to_dttm(self.from_dttm, self.to_dttm)
```

My thought is all these time computation and validation should be 1) reused (it is not currently doesn't mean it can't be) and 2) kept out of the main constructor of `QueryObject` to make the constructor code easy to read.

Otherwise, I'm happy to put the two lines back.

Reply:
I don't feel strongly about indirections but raising on the `get_since_util` SGTM

Reply:
Moving them to all to utils and adding types. I quite like it now. 😄 

```
if time_shift:
        time_shift = parse_human_timedelta(time_shift)
        since = since if since is None else (since - time_shift)
        until = until if until is None else (until - time_shift)

    if since and until and since > until:
        raise ValueError(_('From date cannot be larger than to date'))

    return since, until
```
Then, in the constructor, we simply have
``` 
self.from_dttm, self.to_dttm = utils.get_since_until(time_range, time_shift)
```
Updates coming shortly.